asserts/checkers.go

```diff
@@ -38,12 +39,18 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	return value, nil
 }
 
-func checkAssertType(assertType AssertionType) (*assertionTypeRegistration, error) {
-	reg := typeRegistry[assertType]
-	if reg == nil {
-		return nil, fmt.Errorf("unknown assertion type: %v", assertType)
+var errInvalidAssertionType = errors.New("invalid assertion type")
```

Comment:
ok, now I understand, yes, probably it makes sense to have more informative errors, they are actually more about programming errors now, external inputs can be/are checked with Type()